go/logic/inspect.go

```diff
@@ -190,7 +190,7 @@ func (this *Inspector) inspectOriginalAndGhostTables() (err error) {
 			this.migrationContext.MappedSharedColumns.SetEnumValues(column.Name, column.EnumValues)
 		}
 		if column.Name == mappedColumn.Name && column.Charset != mappedColumn.Charset {
-			this.migrationContext.SharedColumns.SetCharsetConversion(column.Name, column.Charset, mappedColumn.Charset)
+			this.migrationContext.MappedSharedColumns.SetCharsetConversion(column.Name, column.Charset, mappedColumn.Charset)
```

Comment:
@jbielick curious if you can elaborate on the switch from `SharedColumns` to `MappedSharedColumns`. Thanks!

Reply:
Well, chronologically I believe I added this line before #1158 was opened. `SharedColumns` might be the correct usage for #1158 because it leverages `convertArg` and does a conversion of the value in go. I preferred the approach that was similar to the `DatetimetoTimestamp` and `EnumToTextConversion` approaches where the conversion occurs within mysql, as would normally occur if I altered the table without gh-ost.

So it's not that I changed this line, but it now appears in the diff as a change because the base changed and #1158 _also_ introduced this line with a slightly different goal.

I'm using `MappedSharedColumns` because those are the set that allow replacing the "SET x=x" arg with a mysql expression instead of manipulating a raw value itself (which is possible with SharedColumns). We don't need to modify the bytes that are being sent, we need mysql to interpret them differently (and as proposed here convert them).